debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerOffsetContext.java

```diff
@@ -222,7 +222,16 @@ public SqlServerStreamingExecutionState getStreamingExecutionState() {
 
     @Override
     public boolean eventsStreamed() {
-        return (streamingExecutionState.getStatus() == SqlServerStreamingExecutionState.StreamingResultStatus.NO_CHANGES_IN_DATABASE
-                || streamingExecutionState.getStatus() == SqlServerStreamingExecutionState.StreamingResultStatus.NO_MAXIMUM_LSN_RECORDED) == false;
+        if (streamingExecutionState == null) {
```

Comment:
The absence of this condition leads to sporadic test failures caused by a `NullPointerException`:
```
 java.lang.NullPointerException
 	at io.debezium.connector.sqlserver.SqlServerOffsetContext.eventsStreamed(SqlServerOffsetContext.java:225)
 	at io.debezium.pipeline.spi.StreamingResult.eventsStreamed(StreamingResult.java:21)
 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:150)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:834)
```